src/priority.c

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Copyright 2022 Joel E. Anderson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <stumpless.h>
+#include "private/memory.h"
+#include "private/entry.h"
+
+static void toUpperCase( char *str ) {
+  for( int i = 0; i < str[i]; i++) {
```

Comment:
the condition of this for loop appears to be wrong. It should probably check for a NULL character, right?

Reply:
I wander what I thought writing this code.
Done